govcd/nsxv_distributed_firewall.go

```diff
@@ -0,0 +1,501 @@
+/*
+ * Copyright 2023 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// NsxvDistributedFirewall defines a distributed firewall for a NSX-V VDC
+type NsxvDistributedFirewall struct {
+	VdcId         string                       // The ID of the VDC
+	Configuration *types.FirewallConfiguration // The latest firewall configuration
+	Etag          string
+	enabled       bool    // internal flag that signifies whether the firewall is enabled
+	client        *Client // internal usage client
+
+	// Attributes below to be removed if we decide that we don't need services
+	Services      []types.Application      // The list of services for this VDC
```

Comment:
Probably needs to be changed to `Applications` from what I understood in the discussion

Reply:
Sadly, it is not so easy. The API has both `Service` (within the rule structure) and `Application` (as a standalone entity). We can keep things look consistent in the Terraform interface, but to change names in the SDK would be confusing.